### PR TITLE
Added status code, return 422 when trying to delete a busy instance.

### PR DESCRIPTION
**Pull request by TimSimpsonR**
- Added UnprocessableEntity exception from legacy.
- Returned 422 when deleting an instance which is being built.
- Made Instance.delete non-static when I realized that this would simplify the method, give us some validation for free and be almost as fast.
- Created / modified Instance methods "is_sql_running," "status," and "_validate_can_perform_action" by taking code from the legacy InstanceStatus class.

### CODE REVIEW

**Comment by hub-cap** [comment]
+1

File: reddwarf/instance/models.py [hunk 1]
```diff
@@ -96,25 +93,28 @@ def load(context, uuid):
...
-        if uuid is None:
-            raise TypeError("Argument uuid not defined.")
-        LOG.debug("Deleting instance %s..." % uuid)
-        LOG.debug("Creating nova client...")
-        client = create_nova_client(context)
-        instance_info = DBInstance.find_by(id=uuid)
+    def delete(self, force=False):
+        LOG.debug("Deleting instance %s..." % self.id)
+        if not force and self.server.status in SERVER_INVALID_DELETE_STATUSES:
+            raise rd_exceptions.UnprocessableEntity
```
Comment (cp16net) [inline reddwarf/instance/models.py:99]: we should add a message with the id of the instance that can not be processed.

Comment (TimSimpsonR) [inline reddwarf/instance/models.py:99]: We log out the ID above.

Comment (cp16net) [inline reddwarf/instance/models.py:99]: logging and pass back to the user the issue are 2 different things.

Comment (TimSimpsonR) [inline reddwarf/instance/models.py:99]: I was going to say we didn't pass that back but it looks like we did. Committed the change.

Comment (cp16net) [inline reddwarf/instance/models.py:99]: awesome! :)

**Pull request merged**